Deprecate OffscreenCanvas2D's commit() method

Kaiido opened the pull request:
<!-- 🙌 Thanks for contributing to MDN Web Docs. Adding details below will help us to merge your PR faster. -->

### Description

Removes the `OffscreenCanvasContext2D`'s `.commit()` method and fixes little issues in that page.

### Motivation

This method has been removed from the HTML specs in https://github.com/whatwg/html/pull/9979  
Also, while I was on that page I fixed an approximative wording, and the position of the roundRect() method.

### Additional details

The placeholder `<canvas>`'s bitmap is now updated during the "update the rendering" steps of the Worker or Window's event-loop that owns the OffscreenCanvas object. Not sure how to formulate that clearly.

This method has been (partially) implemented in 2 browsers before it got removed, but at least Safari is in the process of removing it (https://github.com/WebKit/WebKit/pull/27196), and there may be another method with the same name coming later. So it's not clear if simply removing the method's page like that is fine or if it would be better to keep it with a deprecation notice instead?

Compat table would also need to be updated, but I'll wait for a position on the above question.

### Related issues and pull requests

<!-- 🔨 If this fully resolves a GitHub issue, use "Fixes #123" -->
<!-- 👉 Highlight related pull requests using "Relates to #123" -->
<!-- ❗ If another pull request should be merged first, use "**Depends on:** #123" -->


<!-- 👷‍♀️ After submitting, go to the "Checks" tab of your PR for the build status -->
`-`

Kaiido committed:
Deprecate OffscreenCanvas2D's commit() method

This method has been removed in https://github.com/whatwg/html/pull/9979

The placeholder <canvas>'s bitmap is now updated during the "update the rendering" steps of the Worker or Window's event-loop that owns the OffscreenCanvas object. Not sure how to formulate that clearly.

This method has been (partially) implemented in 2 browsers before it got removed, but at least Safari is in the process of removing it (https://github.com/WebKit/WebKit/pull/27196), and there may be another method with the same name coming later. So it's not clear if simply removing the method's page like that is fine or if it would be better to keep it with a deprecation notice instead?

Compat table would also need to be updated, but I'll wait for a position on the above question.

Also, while I was on that page I fixed an approximative wording, and the position of the roundRect() method.

skyclouds2001 reviewed (commented):
maybe you should run a shell when delete documentation
see https://developer.mozilla.org/en-US/docs/MDN/Writing_guidelines/Howto/Creating_moving_deleting

skyclouds2001 commented:
> Thanks I wasn't aware of this command, but running it it says there is no file referring to the document and from the git change it seems it doesn't affect anything else. Maybe I'm missing something?
> 
> But as I said in the initial comment, I'm not even sure deleting that page is the right action.

In fact the correct way is to run this command, then the script will automatically delete the file, and do some other works, like checking redirects, modify document meta-info files(_redirects.txt, _wikihistory.json, etc.)

skyclouds2001 commented:
But this seems that, just for this file, no additional work need to do even not running this command

wbamberg commented:
>  But still I'd like confirmation that deleting the page is the right thing to do given the situation, rather than a deprecation notice.

MDN's policy is that once a feature has shipped in a browser, not preffed off by default, then we keep documentation pages for 2 years after the last browser has removed support for it.

I can't find this guideline in MDN's own meta-docs, but in practice it's been adopted from BCD: https://github.com/mdn/browser-compat-data/blob/main/docs/data-guidelines/index.md#removal-of-irrelevant-features.

Kaiido commented:
Alright, then deprecation notice it is. I also opened https://github.com/mdn/browser-compat-data/pull/22878.

Note: I squashed my commits in order to maintain the file's history clean.

bsmth commented:
> I can't find this guideline in MDN's own meta-docs, but in practice it's been adopted from BCD: https://github.com/mdn/browser-compat-data/blob/main/docs/data-guidelines/index.md#removal-of-irrelevant-features.

Dropping a link to the page for reference: https://developer.mozilla.org/en-US/docs/MDN/Writing_guidelines/Experimental_deprecated_obsolete

wbamberg commented:
> > I can't find this guideline in MDN's own meta-docs, but in practice it's been adopted from BCD: https://github.com/mdn/browser-compat-data/blob/main/docs/data-guidelines/index.md#removal-of-irrelevant-features.
> 
> Dropping a link to the page for reference: https://developer.mozilla.org/en-US/docs/MDN/Writing_guidelines/Experimental_deprecated_obsolete

Thanks for the link, Brian! For some reason MDN's metadocs are almost unreadable to me, but AFAICT this does not match our current practice. https://developer.mozilla.org/en-US/docs/MDN/Writing_guidelines/Experimental_deprecated_obsolete#if_the_item_was_implemented_in_a_browser_without_a_flag says:

> It's unlikely that these items will be removed from the documentation anytime soon, if ever.

wbamberg commented on files/en-us/web/api/offscreencanvasrenderingcontext2d/index.md:
```diff
@@ -12,12 +12,12 @@ It is similar to the `CanvasRenderingContext2D` object, with the following diffe
 - there is no support for user-interface features (`drawFocusIfNeeded`, and `scrollPathIntoView`)
 - its `canvas` attribute refers to an `OffscreenCanvas` object rather than a {{HtmlElement("canvas")}} element
-- it has a `commit()` method for pushing rendered images to the context's `OffscreenCanvas` object's placeholder {{HtmlElement("canvas")}} element
+- its `OffscreenCanvas` object's placeholder {{HtmlElement("canvas")}} element's bitmap is updated during the `OffscreenCanvas`'s owner's rendering update.
```
I thought this was a little hard to understand, although this is not easy to phrase.

Maybe:

```suggestion
- the bitmap for the placeholder {{HtmlElement("canvas")}} element belonging to the `OffscreenCanvas` object is updated during the rendering update of the `Window` or `Worker` that owns the `OffscreenCanvas`
```

Is that right?

Kaiido replied:
Yes it's right and a way better formulation. Thanks.

wbamberg commented on files/en-us/web/api/offscreencanvasrenderingcontext2d/commit/index.md:
```diff
@@ -3,14 +3,17 @@ title: "OffscreenCanvasRenderingContext2D: commit() method"
 short-title: commit()
 slug: Web/API/OffscreenCanvasRenderingContext2D/commit
 page-type: web-api-instance-method
+status:
+  - deprecated
+  - non-standard
```
This and the macro calls like `{{Deprecated_Header}}` and `{{Non-standard_Header}}` gets auto-added based on the BCD, so it is better not to include them manually. But it doesn't hurt particularly :).

Kaiido replied:
Shall I also remove the `_Inline` ones from [offscreencanvasrenderingcontext2d/index.md](https://github.com/mdn/content/pull/33145/files/bdcec1a0a4c5d41822d601d8d2aead2ce2d9f9ea#diff-f791ac1361d74aed52e1318331a3e72eec14ffcbffe384fdb516da08bba59ba1)?

Kaiido replied:
@wbamberg gentle ping, 
The info in https://developer.mozilla.org/en-US/docs/MDN/Writing_guidelines/Experimental_deprecated_obsolete#if_the_item_was_implemented_in_a_browser_without_a_flag made me unsure if all these macros, including the ones in the parent page are added automatically or if only the ones on the actual method's page are.

wbamberg replied:
Sorry @Kaiido . Yes, the inline macros are automated as well. There is a page on this at https://developer.mozilla.org/en-US/docs/MDN/Writing_guidelines/Page_structures/Feature_status.

Kaiido committed:
Better wording from suggestion

Co-authored-by: wbamberg <will@bootbonnet.ca>

teoli2003 committed:
Merge branch 'main' into remove-offscreen-commit

Kaiido committed:
remove deprecation macros as per comments

These should hopefully be added back once https://github.com/mdn/browser-compat-data/pull/22878 is merged

wbamberg reviewed (approved):
👍 thank you @Kaiido !

Kaiido commented:
Thanks to all the reviewers for guiding me through my first deprecation notice.
